### PR TITLE
Fix: errors are reported in test case for junit

### PR DESCRIPTION
**Pull request by Vasilii-Besschetnov**
- Error list is not reset by reference anymore
- All errors are include in the report, not only the first one

Thanks in advance for your contribution. Please follow the below steps in submitting a pull request, as it will help us with addressing it quicker.

- [ ] Create a new branch from master (e.g. `features/my-new-feature` or `issue/123-my-bugfix`)
- [ ] If you're fixing a bug also create an issue if one doesn't exist yet
- [ ] If it's a new feature explain why do you think it's necessary
- [ ] If your change include drastic or low level changes please discuss them to make sure they will be accepted and what the impact will be
- [ ] If your change is based on existing functionality please consider refactoring first. Pull requests that duplicate code will not make it in very quick, if at all.
- [ ] Do not include changes that are not related to the issue at hand
- [ ] Follow the same coding style with regards to spaces, semicolons, variable naming etc.
- [ ] Add unit tests

### CODE REVIEW

File: lib/runner/reporters/junit.xml.ejs [hunk 1]
```diff
@@ -8,16 +8,18 @@
...
     <testcase name="<%= item %>" classname="<%= className %>" time="<%= testcase.time %>" assertions="<%= assertions.length %>"><%
       for (var i = 0; i < assertions.length; i++) { %><% if (assertions[i].failure) { %>  <failure message="<%= assertions[i].message %>"><%= assertions[i].stackTrace %></failure><% } %>
 <% if (assertions[i].screenshots && assertions[i].screenshots.length > 0) { %><system-out><% for (var j = 0; j < assertions[i].screenshots.length; j++) { %>[[ATTACHMENT|<%= assertions[i].screenshots[j] %>]]<% } %></system-out><% } %>
     <% }
     if (testcase.failed > 0 && testcase.stackTrace) {
     %><failure message="<%= testcase.message %>"><%= testcase.stackTrace %></failure><% }
-    if (testcase.errors > 0 && testcase.errorStackTrace) {
-    %>
-    <error message="Error executing test code"><%= testcase.errorStackTrace %></error><% } %>
+    for (var i = 0; i < errorMessages.length; i++) { %>
```
Comment (laurentgo) [inline lib/runner/reporters/junit.xml.ejs]: According to the schema, you can have at most 1 element (note that you cannot have both error and failure too, so should be some kind of else if...)

Comment (Vasilii-Besschetnov) [reply lib/runner/reporters/junit.xml.ejs]: if I right understand the xsd schema, there are no any limitation how many errors and failures could occur.
<xs:choice minOccurs="0">
  <xs:element name="error">...</xs:element>
  <xs:element name="failure"></xs:element>
</xs:choice>
I read that schema as "there could be 0 or more occurrences of elements, that appear in choice element. That means there could be any number of failure and errors.
Did I misunderstand the schema?

Comment (Vasilii-Besschetnov) [reply lib/runner/reporters/junit.xml.ejs]: Please disregard my comment. I thought that maxoccurs is defaulted to unbounded.

Comment (Vasilii-Besschetnov) [reply lib/runner/reporters/junit.xml.ejs]: I altered a code to generate a single error element only if failure does not exist

**Review by laurentgo — APPROVED** [review]

**Pull request merged**